dashboard/app/models/user.rb

```diff
@@ -786,11 +812,19 @@ def track_level_progress(script_level, new_result)
       user_level.save!
     end
 
-    if new_level_completed && script_level.script
-      track_script_progress(script_level.script)
+    if new_level_completed && script_id
+      User.track_script_progress(user_id, script_id)
     end
+  end
 
-    new_level_completed
+  def User.handle_async_op(op)
+    raise 'Model must be User' if op['model'] != 'User'
+    case op['action']
+      when 'track_level_progress'
+        User.track_level_progress_sync(op['user_id'], op['level_id'], op['script_id'], op['new_result'])
```

Comment:
I prefer explicitness in this case.  I expect to have few if any more async user ops so it's not a big burden writing this.